pact/kip/poly-fungible-v2.pact

```diff
@@ -0,0 +1,177 @@
+  )
+
+  (defcap TOKEN:bool (id:string)
+    @doc " Emitted when token ID is created."
+    @event
+  )
+
+  (defun precision:integer (id:string)
+    @doc
+      " Return maximum decimal precision for ID."
+  )
+
+  (defun enforce-unit:bool
+    ( id:string
+      amount:decimal
+    )
+    @doc
+      " Enforce that AMOUNT meets minimum precision allowed for ID."
+  )
+
+  (defun create-account:string
+    ( id:string
+      account:string
+      guard:guard
+    )
+    @doc
+      " Create ACCOUNT for ID with 0.0 balance, with GUARD controlling access."
+    @model
+      [ (property (!= id ""))
+        (property (!= account ""))
+      ]
+  )
+
+  (defun get-balance:decimal
+    ( id:string
+      account:string
+    )
+    @doc
+      " Get balance of ID for ACCOUNT. Fails if account does not exist."
+  )
+
+  (defun details:object{account-details}
+    ( id:string
+      account:string
+    )
+    @doc
+      " Get details of ACCOUNT under ID. Fails if account does not exist."
+  )
+
+  (defun rotate:string
+    ( id:string
+      account:string
+      new-guard:guard )
+    @doc
+      " Rotate guard for ACCOUNT for ID to NEW-GUARD, validating against existing guard."
+    @model
+      [ (property (!= id ""))
+        (property (!= account ""))
+      ]
+
+  )
+
+  (defun transfer:string
+    ( id:string
+      sender:string
+      receiver:string
+      amount:decimal
+    )
+    @doc
+      " Transfer AMOUNT of ID between accounts SENDER and RECEIVER. \
+      \ Fails if SENDER does not exist. Managed by TRANSFER."
+    @model
+      [ (property (> amount 0.0))
+        (property (!= id ""))
+        (property (!= sender ""))
+        (property (!= receiver ""))
+        (property (!= sender receiver))
+      ]
+  )
+
+  (defun transfer-create:string
+    ( id:string
+      sender:string
+      receiver:string
+      receiver-guard:guard
+      amount:decimal
+    )
+    @doc
+      " Transfer AMOUNT of ID between accounts SENDER and RECEIVER. \
+      \ If RECEIVER exists, RECEIVER-GUARD must match existing guard; \
+      \ if RECEIVER does not exist, account is created. \
+      \ Managed by TRANSFER."
+    @model
+      [ (property (> amount 0.0))
+        (property (!= id ""))
+        (property (!= sender ""))
+        (property (!= receiver ""))
+        (property (!= sender receiver))
+      ]
+  )
+
+  (defpact transfer-crosschain:string
+    ( id:string
+      sender:string
+      receiver:string
+      receiver-guard:guard
+      target-chain:string
+      amount:decimal
+    )
+    @doc
+      " Transfer AMOUNT of ID between accounts SENDER on source chain \
+      \ and RECEIVER on TARGET-CHAIN. If RECEIVER exists, RECEIVER-GUARD \
+      \ must match existing guard. If RECEIVER does not exist, account is created."
+    @model
+      [ (property (> amount 0.0))
+        (property (!= id ""))
+        (property (!= sender ""))
+        (property (!= receiver ""))
+        (property (!= target-chain ""))
+      ]
+  )
+
```

Comment:
Add `update-supply`?

Reply:
No b/c that is a private/implementation-specific function.